### ISSUE TITLE
Allow deposit by someone who is not a token owner

### ISSUE THREAD

**Issue by bogdan** [maticnetwork/pos-portal#101]
## Motivation

In our app we want to provide people a better user experience when they deposit their tokens to polygon.
That is why we want `RootChainManager#depositFor` to be callable by an address that is not a token owner.

## Problem

That is not currently possible with `MintableERC721Predicate` (I didn't check others, but it is probably the same issue there).

Here is why:
`RootChainManager` always uses `_msgSender` as a `depositor`: https://github.com/maticnetwork/pos-portal/blob/88dbf0a88fd68fa11f7a3b9d36629930f6b93a05/contracts/root/RootChainManager/RootChainManager.sol#L313
`depositor` is always passed as a first argument to `ERC721#safeTransferFrom` which should always be a token owner: https://github.com/maticnetwork/pos-portal/blob/88dbf0a88fd68fa11f7a3b9d36629930f6b93a05/contracts/root/TokenPredicates/MintableERC721Predicate.sol#L94
Additionally `depositor` is also emitted as a part of `LockedMintableERC721` event which I believe is how polygon node knows who should be a token owner on polygon: https://github.com/maticnetwork/pos-portal/blob/88dbf0a88fd68fa11f7a3b9d36629930f6b93a05/contracts/root/TokenPredicates/MintableERC721Predicate.sol#L90

## Solution

Use `ERC721#ownerOf` instead of `depositor` to emit `LockedMintableERC721` and calling `ERC721#safeTransferFrom`.

If that is not acceptable, can you explain a motivation of this limitation?

**Comment by itzmeanjan** [comment]
Hello @bogdan , given that https://github.com/OpenZeppelin/openzeppelin-contracts/blob/4961a51cc736c7d4aa9bd2e11e4cbbaff73efee9/contracts/token/ERC721/ERC721.sol#L181 kind check is implemented in root contract and `depositor` is non-owner address but allowed to transfer on owner's behalf, does it not solve your problem ?

**Comment by bogdan** [comment]
@itzmeanjan nope, here the problem is in the `from` argument of `safeTransferFrom` that in the should be exactly equal to token owner: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/4961a51cc736c7d4aa9bd2e11e4cbbaff73efee9/contracts/token/ERC721/ERC721.sol#L332 all the time.

**Comment by itzmeanjan** [comment]
Yes, I get it now. For ERC721 we can probably implement it.

**Comment by itzmeanjan** [comment]
Would you be able to send one PR @bogdan ?
